cyclonedx/model/component.py

```diff
@@ -477,6 +480,37 @@ def has_vulnerabilities(self) -> bool:
         """
         return bool(self.get_vulnerabilities())
 
+    def has_dependencies(self) -> bool:
+        """
+        Does this Component have any dependencies?
+
+        Returns:
+             `True` if this Component has 1 or more dependency, `False` otherwise.
+        """
+        return bool(self.__dependencies)
+
+    def get_dependencies(self) -> List[Dependency]:
+        """
+        Get all the Dependencies for this Component.
```

Comment:
We've been trying to utilise the descriptions for attributes as provided in the CycloneDX Specification - this doc would be:

> Defines the direct dependencies of a component. Components that do not have their own dependencies MUST be declared as empty elements within the graph. Components that are not represented in the dependency graph MAY have unknown dependencies. It is RECOMMENDED that implementations assume this to be opaque and not an indicator of a component being dependency-free.